Formula/ledger.rb

```diff
@@ -24,9 +24,10 @@ class Ledger < Formula
 
   depends_on "cmake" => :build
   depends_on "boost"
+  depends_on "boost-python3"
   depends_on "gmp"
   depends_on "mpfr"
-  depends_on "python@3.10"
+  depends_on "python@3.9"
```

Comment:
We should wait until `boost-python3` is migrated to `python@3.10` instead of switching this back to `python@3.9`.

Reply:
I created #89179 (although ledger currently doesn't actually use the Python 3.9 dependency at all AFAICT).

Reply:
Never mind. I realize now that because of all the dependencies updating `boost-python3` is equivalent to updating everything to Python 3.10. There is probably someone more qualified than me to do that.

Right now this formula lists Python as a dependency without actually using it though, so it's not really switching back. It will just become part of whatever large PR has to update all the formulae that depend on `boost-python3`. And in the meantime it at least fixes Ledger to have Python support again.